### ISSUE TITLE
JetBrains Annotations: Part II

### ISSUE THREAD

**Issue by TAGC** [pomma89/Thrower#4]
Thanks again for the annotating your methods with `[NoEnumeration]`. If possible, I'd like to expand on that to further support integration with Resharper.

One improvement that would be really nice would be to annotate the null guard methods, like this:

```cs
[ContractAnnotation("argument:null => halt")]
public void IfIsNull<TArg>(TArg argument, string argumentName, string message = null)
{
    // ...
}
```

For methods that only take a single parameter, the contract can be simplified:

```cs
[ContractAnnotation("null => halt")]
public void IfIsNull<TArg>(TArg argument)
{
    // ...
}
```

I'm still unable to open the Pommalabs.Thrower solution reliably without Visual Studio crashing, so I can't submit a pull request. However, I've tested these changes with simple analogs of some of your methods:

![2017-09-18_11-27-38](https://user-images.githubusercontent.com/1588951/30538172-a31e3574-9c64-11e7-8131-458d0b4aceeb.gif)

**Comment by pomma89** [comment]
Hi @TAGC,

Your idea is interesting and I will add it as soon as possible in the following days.

Thanks!

**Comment by pomma89** [comment]
Hi @TAGC,

Support for contract attribute should be avalaible in v4.2.2. Can you confirm it is OK? I added support for null clause, but I also added it for some methods like If/IfNot (I hope they are OK).

Thanks again!

**Comment by TAGC** [comment]
Yep, looks great. And it was a really good idea annotating the `If` and `IfNot` methods too; I didn't think of that.

![2017-09-21_09-32-18](https://user-images.githubusercontent.com/1588951/30686371-f6c4e90a-9eaf-11e7-949a-5d3ccf6040ec.gif)

I've reviewed your changes and it all looks right, and it works at least for this snippet of code. 👍

**Comment by TAGC** [comment]
Oh, not that it's a huge thing either way but I think those annotations would also fit [here](https://github.com/pomma89/Thrower/blob/dev-pomma89/src/PommaLabs.Thrower/ExceptionHandlers/ArgumentExceptionHandler.cs), for things like `IsNullOrWhitespace`, etc.

**Comment by pomma89** [comment]
Hi @TAGC,

Missing annotations should be available in v4.2.3, released just now.

Thanks!
